### PR TITLE
API "auth_token" authentication support

### PR DESCRIPTION
**Pull request by jathanism**
This is implements authentication  for NSoT. 

Highlights:

+ Cookies are stored as secure cookies using cookie_secret.
+ Return correct codes for auth handlers
+ Web views use "default" auth (currently user_auth_header)
+ API views use "default" or "auth_token"
+ AuthToken can be done using "Authorization" header or query args.
+ Unit tests for new API auth functionality!

### CODE REVIEW

**Comment by jathanism** [comment]
FYI this latest build failed for a reason unrelated to the testing (died during virtualenv construction). I triggered build 97 to run again.

**Comment by jathanism** [comment]
The latest 2 commits (77195c8 and aea9976) have this almost ready for merge. Still need to add the updates to the docs.

**Comment by jathanism** [comment]
I updated the documentation and rebased the commit for this PR. If there are no objections, it's read to go.

File: docs/conf.py [hunk 1]
```diff
@@ -49,16 +49,19 @@
...
+copyright = u'2014-2015, Dropbox, Inc.'
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = '0.0.1'
+def get_version():
+    from nsot import __version__
```
Comment (gmjosack) [inline docs/conf.py]: This doesn't return the __version__

Comment (jathanism) [inline docs/conf.py]: LOL. Derp.

File: nsot/models.py [hunk 2]
```diff
@@ -199,6 +221,9 @@ class User(Model):
     id = Column(Integer, primary_key=True)
     email = Column(String(length=255), unique=True, nullable=False)
 
+    # The user's secret key is used to generate their auth_toekn
```
Comment (gmjosack) [inline nsot/models.py]: s/toekn/token/

**Pull request merged**